src/wfs/src/test/java/org/geoserver/wfs/response/CSVOutputFormatTest.java

```diff
@@ -225,4 +228,59 @@ public void testFullRequestWithDynamicCsvSeparator() throws Exception {
             assertEquals(fs.getSchema().getDescriptors().size() + 1, line.length);
         }
     }
+
+    @Test
+    public void testDoubleQuotesAsCsvSeparator() throws Exception {
+
+        // build the request objects and feed the output format
+        GetFeatureType gft = WfsFactory.eINSTANCE.createGetFeatureType();
+        HashMap<String, String> hashMap = new HashMap<>();
+        hashMap.put("CSVSEPARATOR", "\"");
+        gft.setFormatOptions(hashMap);
+        Operation op =
+                new Operation("GetFeature", getServiceDescriptor10(), null, new Object[] {gft});
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        FeatureCollectionResponse fct =
+                FeatureCollectionResponse.adapt(WfsFactory.eINSTANCE.createFeatureCollectionType());
+
+        SimpleFeatureTypeBuilder builder = new SimpleFeatureTypeBuilder();
```

Comment:
The feature looks more complicated than it needs to be (probably because of copy-n-paste - perhaps that could be refactored), but I guess its still a valid test.

Reply:
Refactored. (used mock data instead of creating it)